openml/runs/functions.py

```diff
@@ -349,82 +361,93 @@ def _run_task_get_arffcontent(model, task, class_labels):
-            train_indices, test_indices = fold
-            trainX = X[train_indices]
-            trainY = Y[train_indices]
-            testX = X[test_indices]
-            testY = Y[test_indices]
-
-            try:
-                # for measuring runtime. Only available since Python 3.3
-                if can_measure_runtime:
-                    modelfit_starttime = time.process_time()
-                model_fold.fit(trainX, trainY)
-
-                if can_measure_runtime:
-                    modelfit_duration = (time.process_time() - modelfit_starttime) * 1000
-                    user_defined_measures['usercpu_time_millis_training'][rep_no][fold_no] = modelfit_duration
-            except AttributeError as e:
-                # typically happens when training a regressor on classification task
-                raise PyOpenMLError(str(e))
-            
-            # extract trace, if applicable
-            if isinstance(model_fold, sklearn.model_selection._search.BaseSearchCV):
-                arff_tracecontent.extend(_extract_arfftrace(model_fold, rep_no, fold_no))
-
-            # search for model classes_ (might differ depending on modeltype)
-            # first, pipelines are a special case (these don't have a classes_
-            # object, but rather borrows it from the last step. We do this manually,
-            # because of the BaseSearch check)
-            if isinstance(model_fold, sklearn.pipeline.Pipeline):
-                used_estimator = model_fold.steps[-1][-1]
-            else:
-                used_estimator = model_fold
+    num_reps, num_folds, num_samples = task.get_split_dimensions()
+
+    for rep_no in range(num_reps):
+        for fold_no in range(num_folds):
+            for sample_no in range(num_samples):
+                model_fold = sklearn.base.clone(model, safe=True)
+                train_indices, test_indices = task.get_train_test_split_indices(repeat=rep_no,
+                                                                                fold=fold_no,
+                                                                                sample=sample_no)
+                trainX = X[train_indices]
+                trainY = Y[train_indices]
+                testX = X[test_indices]
+                testY = Y[test_indices]
+
+                try:
+                    # for measuring runtime. Only available since Python 3.3
+                    if can_measure_runtime:
+                        modelfit_starttime = time.process_time()
+                    model_fold.fit(trainX, trainY)
+
+                    if can_measure_runtime:
+                        modelfit_duration = (time.process_time() - modelfit_starttime) * 1000
+                        user_defined_measures_sample['usercpu_time_millis_training'][rep_no][fold_no][sample_no] = modelfit_duration
+                        user_defined_measures_fold['usercpu_time_millis_training'][rep_no][fold_no] = modelfit_duration
```

Comment:
Yes. In the case of measures per fold, we are interested in how something ran on the full dataset. Which is enforced by the way that the last loop is always the result on the full dataset